lib/net/websocket_server.js

```diff
@@ -166,7 +166,22 @@ function ws (instance, change, options, callback) {
       return null
     })
 
-    instance.on(changeEvent, function (changes) {
+    socket.on('close', function () {
+      instance.off(changeEvent, changeListener)
+      process.removeListener('message', ipcListener)
```

Comment:
I think that this may need to be qualified with `if (isWorker && useIPC)`, but there's no harm in removing a listener that doesn't exist anyways.